addons/product_expiry/models/production_lot.py

```diff
@@ -95,7 +95,7 @@ def _alert_date_exceeded(self):
         for lot in alert_lots:
             lot.activity_schedule(
                 'product_expiry.mail_activity_type_alert_date_reached',
-                user_id=lot.product_id.responsible_id.id or SUPERUSER_ID,
+                user_id=lot.product_id.with_company(lot.company_id).responsible_id.id or SUPERUSER_ID,
```

Comment:
Your fix is good for the bugfix's specific use case, but it may change behavior for some users since they may already be receiving alerts based on only the first company's responsible being assigned and not the other companies (i.e. expired lots in other companies may suddenly start only having alerts for Odoobot for some products). I suggest you keep the current behavior as backup in case they do not have a responsible assigned to specific lot's company

```suggestion
                user_id=lot.product_id.with_company(lot.company_id).responsible_id.id or lot.product_id.responsible_id.id or SUPERUSER_ID,
```

Reply:
Indeed, that's a good idea! Thank you for your review :slightly_smiling_face: